builtin/core.wren

```diff
@@ -21,6 +21,16 @@ class Sequence {
     }
     return true
   }
+
+  reduce(acc, f) {
+    for (element in this) {
+      acc = f.call(acc, element)
+    }
+    return acc
+  }
+
+  reduce(f) { this[1..-1].reduce(this[0], f) }
```

Comment:
I initially had the accumulator behind the function, like in Haskell: `foldl :: (a -> b -> a) -> a -> [b] -> a` (i.e. first the function, then the accumulator, then the list), but not being able to use a block argument would be bad.

@munificent: Your version of reduce isn't as bad as I feared, I wasn't aware of the underlying iterator methods :)